hal/inc/hal_platform.h

```diff
@@ -207,6 +211,10 @@
 #define HAL_PLATFORM_SPI_DMA_SOURCE_RAM_ONLY (0)
 #endif // HAL_PLATFORM_SPI_DMA_SOURCE_RAM_ONLY
 
+#ifndef HAL_PLATFORM_SPI_HAL_THREAD_SAFETY
+#define HAL_PLATFORM_SPI_HAL_THREAD_SAFETY (0)
```

Comment:
Defaulting to `(1)`, true, and disabling for `PLATFORM_GCC` specifically.

Reply:
After discussing, we decided to default this value to zero, and enable where appropriate.